presto-main/src/test/java/com/facebook/presto/type/TestJsonOperators.java

```diff
@@ -115,6 +116,34 @@ public void testCastFromDouble()
     }
 
     @Test
+    public void testCastToReal()
+            throws Exception
+    {
+        assertFunction("cast(JSON 'null' as REAL)", REAL, null);
+        assertFunction("cast(JSON '-128' as REAL)", REAL, -128.0f);
+        assertFunction("cast(JSON '128' as REAL)", REAL, 128.0f);
+        assertFunction("cast(JSON '1.234567890' as REAL)", REAL, 1.234567890f);
+        assertFunction("cast(JSON '128.9' as REAL)", REAL, 128.9f);
+        assertFunction("cast(JSON '1e-324' as REAL)", REAL, 0.0f); // smaller than minimum subnormal positive
+        assertFunction("cast(JSON '1e309' as REAL)", REAL, Float.POSITIVE_INFINITY); // overflow
+        assertFunction("cast(JSON '-1e309' as REAL)", REAL, Float.NEGATIVE_INFINITY); // underflow
+        assertFunction("cast(JSON 'true' as REAL)", REAL, 1.0f);
+        assertFunction("cast(JSON 'false' as REAL)", REAL, 0.0f);
+        assertFunction("cast(JSON '\"128\"' as REAL)", REAL, 128.0f);
+        assertFunction("cast(JSON '\"12345678901234567890\"' as REAL)", REAL, 1.2345678901234567e19f);
```

Comment:
@haozhun The goal of this test is to test when the underlying integer is overflowing BIGINT, it should still successfully convert into a REAL number although with precision loss.